### DISCUSSION TITLE
Issues with iaikPkcs11Provider

### DISCUSSION THREAD

**Discussion by mvoucko** [Keyfactor/ejbca-ce#163] [Q&A]
For testing I added iaik_jce, iaikPkcs11Provider, iaikPkcs11Wrapper as described here

  *** SUN or IAIK PKCS#11 Provider ***
  (https://download.primekey.se/docs/EJBCA-Enterprise/6_7_0/adminguide.html)

and used pkcs11-spy to access a softhsm2 file based token. 

Testing the key contained in the softhsm withth clientToolBox works fine, setting up the token via the webinterface as well. But when testing the configured cryptoToken via the webinterface the following error shows up 

  Error: java.lang.UnsupportedOperationException: Private Exponent value is either not present or sensitive.

I found a post from 2013

  https://sourceforge.net/p/ejbca/mailman/ejbca-develop/thread/F007130F6FFEE243B03C55BDBA08A27DB316A0%40BUMBX01WP.fr.ad.bull.net/#msg30395021

which correctly states this is a problem with iaikPkcs11Provider implementation but unfortunately I can not find any solution to this issue.

Does it mean it is not possible to use the iaik stack at all or is this an issue with the community edition only?

Any link, tip or hint is highly appreciated.

Thanks
 Michael

**Comment by primetomas** [comment]
Hi,
Usage of IAIK in EJBCA has not been tested for probably more than a decade. Although there are traces remaining of it in the code, it is not maintained or supported.
Is there a specific reason you are looking at using that?

Cheers,
Tomas
